src/main/java/cubicchunks/asm/mixin/core/MixinWorld_HeightLimits.java

```diff
@@ -166,10 +166,10 @@ private void isAreaLoadedInject(int xStart, int yStart, int zStart, int xEnd, in
 		if (!this.isCubicWorld()) {
 			return;
 		}
-		boolean ret = this.blocksExist(
-				xStart, yStart, zStart,
-				xEnd, yEnd, zEnd,
-				allowEmpty, GeneratorStage.LIGHTING
+		// TODO: Checking for GeneratorStage.LIVE instead of lighting, because this is the only publicly available stage. Analyze effects.
```

Comment:
isAreaLoaded should return true of these blocks exist. As far as I know isAreaLoaded itself is used when calculating lighting. When isAreaLoaded returns false when trying to calculate lighting - it's added to a queue and calculated outside of generation pipeline. So it should actually check for lighting, or something a bit earlier. Maybe adding isInitialStage method and changing the blocksExist to accept Predicate would be a good idea?